ovs/lib/albaupdate.py

```diff
@@ -468,75 +297,81 @@ def _package_install_sdm(package_info, components=None):
         :param components: Update components which have been executed
         :type components: list
+        :param update_information: Information required for an update (defaults to None for backwards compatibility)
+        :type update_information: dict
         :return: None
+        :rtype: NoneType
         """
-        if 'framework' not in components and 'alba' not in components:
-            return
+        method_name = inspect.currentframe().f_code.co_name
+        cls._logger.info('{0}: Executing hook {1}'.format(client.ip, method_name))
+        pkg_names_to_check = set()
+        for component, package_names in PackageFactory.get_package_info()['names'].iteritems():
+            if component in components:
+                pkg_names_to_check.update(package_names)
 
-        from ovs_extensions.generic.toolbox import ExtensionsToolbox
-
-        update_information = AlbaUpdateController._get_update_information_alba_plugin({})
-        services_to_restart = set()
-        if 'alba' in components:
-            services_to_restart.update(update_information.get('alba', {}).get('services_post_update', set()))
-        if 'framework' in components:
-            services_to_restart.update(update_information.get('framework', {}).get('services_post_update', set()))
-
-        # Restart Arakoon (and other services)
-        if services_to_restart:
-            AlbaUpdateController._logger.info('{0}: Executing hook {1}'.format(client.ip, inspect.currentframe().f_code.co_name))
-            for service_name in sorted(services_to_restart):
-                if not service_name.startswith('arakoon-'):
-                    UpdateController.change_services_state(services=[service_name], ssh_clients=[client], action='restart')
-                else:
-                    cluster_name = ArakoonClusterConfig.get_cluster_name(internal_name=ExtensionsToolbox.remove_prefix(service_name, 'arakoon-'))
-                    master_ip = StorageRouterList.get_masters()[0].ip if cluster_name == 'config' else None
-                    temp_cluster_name = 'cacc' if cluster_name == 'config' else cluster_name
-                    try:
-                        arakoon_metadata = ArakoonInstaller.get_arakoon_metadata_by_cluster_name(cluster_name=temp_cluster_name,
-                                                                                                 ip=master_ip)
-                    except ArakoonNoMaster:
-                        AlbaUpdateController._logger.warning('Arakoon cluster {0} does not have a master, not restarting related services'.format(cluster_name))
-                        continue
-                    except ArakoonNotFound:
-                        AlbaUpdateController._logger.warning('Arakoon cluster {0} does not have the required metadata key, not restarting related services'.format(cluster_name))
-                        continue
+        try:
+            ServiceFactory.remove_services_marked_for_removal(client=client,
```

Comment:
Is this something we can easily ignore?
What is the impact of not removing the services marked for removal?

Reply:
We introduced this when we renamed the albaproxy services (When multiple proxies were required per SD)
So without the removal, the old services will remain running, but shouldn't do anything since they aren't configured anymore